New readme

samhumeau opened the pull request:
**Patch description**
Revamp of the Readme, clearer, shorter.

jaseweston commented:
- i think the list of tasks could be better at the beginning, maybe saying how many there are at the beginning

- mention zoo at the beginning too? 

- and research/projects page?

- i think if we could replace s2s example with transformer could be nice, and not on babi

- the code structure (link) could be a link to ?:   https://github.com/facebookresearch/ParlAI/tree/new_readme/parlai

- the detailed goals also doesnt seem to fit now? i don't know .. it looks very wordy because nothing else is wordy

emilydinan commented:
I think we should update the examples or take them out. @hadasah may have opinions after writing the quick start.

hadasah commented on README.md:
```diff
@@ -4,326 +4,117 @@
+ - [List of available tasks/datasets](https://parl.ai/docs/tasks.html)
+ - [Creating a dataset/task](http://www.parl.ai/docs/tutorial_task.html).
+ - [List of available agents](./parlai/agents)
+ - [Creating a new agent](https://parl.ai/docs/tutorial_seq2seq.html#).
```
Nit: remove periods for this and the line two above.

hadasah commented on README.md:
```diff
@@ -4,326 +4,117 @@
+## Documentation
 
-## Properties
+ - [Basics: world, agents, teachers, action and observations](https://parl.ai/docs/tutorial_basic.html)
```
I think we probably want to add the Quick Start too.

hadasah commented on README.md:
```diff
@@ -4,326 +4,117 @@
-SQuAD, bAbI tasks, MCTest, WikiQA, WebQuestions, SimpleQuestions, WikiMovies, QACNN, QADailyMail, CBT, BookTest, bAbI Dialog tasks,
-Ubuntu, OpenSubtitles, Cornell Movie, VQA-COCO2014.
-Since then, several datasets have been added such as  VQAv2, VisDial, MNIST_QA, Personalized Dialog, InsuranceQA, MS MARCO, TriviaQA, and CLEVR. See [here](https://github.com/facebookresearch/ParlAI/blob/master/parlai/tasks/task_list.py) for the current complete task list.
+## Detailed Goals
```
Yeah, having 3 or 4 separate bulleted lists under this is really awkward. I'm not sure we should have this section at all - but if we do, we should condense it better.

samhumeau replied:
I removed that section, added a few stuff in the first paragraph

hadasah commented on README.md:
```diff
@@ -4,326 +4,117 @@
-- Supports batch and hogwild training and evaluation of models.
 
-## Basic Examples
+## Examples
```
Yeah, we talked about having examples that are more similar to what users will likely actually use ParlAI for. Maybe we should just grab the display_data, train_model, and eval_model from the transformer on twitter section.

samhumeau replied:
This is a good idea, this is an actual good example

pringshia commented on README.md:
```diff
@@ -4,326 +4,117 @@
-Our aim is for the number of tasks and agents that train on them to grow in a community-based way.
+- a **unified framework** for sharing, training and testing dialogue models from open-domain chitchat to VQA.
+- many popular **datasets available all in one place, with the same API**, among them [SQuAD](https://rajpurkar.github.io/SQuAD-explorer/), [bAbI tasks](https://arxiv.org/abs/1502.05698), [MS MARCO](http://www.msmarco.org/), [WikiQA](https://www.microsoft.com/en-us/download/details.aspx?id=52419), [SimpleQuestions](https://arxiv.org/abs/1506.02075), [WikiMovies](https://arxiv.org/abs/1606.03126), ...
+- a wide set of **reference models** -- from retrieval baselines to Transformers.
```
Is Transformers supposed to be capitalized as such?

pringshia commented on README.md:
```diff
@@ -4,326 +4,117 @@
+This is the recommended installation procedure, as it provides ready access to the examples and allows you to modify anything you might need. This is especially useful if you if you want to submit another task to the repository.
 
-Set of datasets to bootstrap a working dialogue model for human interaction
+All needed data will be downloaded to ~/ParlAI/data, and any non-data files (such as the MemNN code) if requested will be downloaded to ~/ParlAI/downloads. If you need to clear out the space used by these files, you can safely delete these directories and any files needed will be downloaded again.
```
What is the MemNN code? Perhaps this should be a link for reference? The qualification here is nice, but a way to get more context for beginners would be nice.

pringshia replied:
I also find it easier to read when file paths are wrapped in code blocks, e.g. `~/ParlAI/downloads`... I notice this in some places like here:

https://github.com/facebookresearch/ParlAI/pull/1748/files#diff-04c6e90faac2675aa89e2176d2eec7d8R89

Maybe we can do it everywhere.

pringshia commented on README.md:
```diff
@@ -4,326 +4,117 @@
-Included are examples of training neural models with [PyTorch](http://pytorch.org/), with batch training on GPU or hogwild training on CPUs. Using [Tensorflow](https://www.tensorflow.org/) or other frameworks instead is also straightforward.
-
-Our aim is for the number of tasks and agents that train on them to grow in a community-based way.
+- a **unified framework** for sharing, training and testing dialogue models from open-domain chitchat to VQA.
```
This bullet point also seems a little repetitive given the first sentence, perhaps we could somehow merge this bullet right into the first sentence:

> ParlAI (pronounced “par-lay”) is a framework for dialogue AI research, implemented in Python.

into 

> ParlAI (pronounced “par-lay”) is a **unified framework** for dialogue AI research, implemented in Python. It provides a platform for sharing, training and testing dialogue models &mdash; from open-domain chitchat to VQA (visual question answering).

pringshia replied:
Potential buzzword we can use: "[batteries](https://www.python.org/dev/peps/pep-0206/#batteries-included-philosophy)-[included](https://www.quora.com/What-does-batteries-included-philosophy-mean)"

samhumeau commented:
@jaseweston  :
"I'm not sure the "Basics: world, agents, teachers, action and observations" actually explains the way worlds and main loop works is explained as well as the original README, so we need some work there"
So Margaret have been remixing these in https://github.com/facebookresearch/ParlAI/pull/1731/files

samhumeau commented:
Requesting another review @stephenroller  @jaseweston @hadasah @pringshia :

- removed detailed goals an property but added some stuff in the main list of gains in first part
- changed the examples for transformer ones
- nits and bits correction

emilydinan commented:
@samhumeau  is this ready to merge

samhumeau commented:
I think so I was waiting for opinions
